pkg/operator/reporting/templates_test.go

```diff
@@ -0,0 +1,525 @@
+	metering "github.com/operator-framework/operator-metering/pkg/apis/metering/v1alpha1"
+	"github.com/operator-framework/operator-metering/pkg/presto"
+	"github.com/operator-framework/operator-metering/test/testhelpers"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	testNamespace            = "default"
+	testReportName           = "unready-deployment-replicas"
+	testReportQuery          = "unready-deployment-replicas"
+	testInvalidQueryName     = "invalid-query"
+	testNonExistentQueryName = "does-not-exist"
+	testCatalogName          = "hive"
+	testTimeString           = "2019-05-01T15:00:05Z"
+)
+
+func TestRenderQuery(t *testing.T) {
+	reportStart := &time.Time{}
+	reportEndTmp := reportStart.AddDate(0, 1, 0)
+	reportEnd := &reportEndTmp
+
+	ds1 := testhelpers.NewReportDataSource("datasource1", testNamespace)
+	ds1.Status.TableRef = v1.LocalObjectReference{Name: "initialized_datasource"}
+	ds2 := testhelpers.NewReportDataSource("datasource2", testNamespace)
+
+	testValidQuery := &metering.ReportQuery{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testReportQuery,
+			Namespace: testNamespace,
+		},
+		Spec: metering.ReportQuerySpec{
+			Inputs: []metering.ReportQueryInputDefinition{
+				{
+					Name:     testReportName,
+					Type:     "ReportDataSource",
+					Required: true,
+					Default:  newDefault((`"` + ds1.Name + `"`)),
+				},
+			},
+			Query: "SELECT * FROM test_table",
+		},
+	}
+
+	testTable := []struct {
+		name            string
+		reportTemplate  *ReportQueryTemplateContext
+		templateContext TemplateContext
+		expectErr       bool
+		expectErrMsg    string
+		expectOutput    string
+	}{
+		{
+			name: "valid report query with no templating returns nil and the expected query output",
+			reportTemplate: &ReportQueryTemplateContext{
+				ReportQuery: newTestReportQuery(ds1, "show tables;"),
+			},
+			templateContext: TemplateContext{
+				Report: ReportTemplateInfo{},
+			},
+			expectOutput: "show tables;",
+		},
+		{
+			name: "valid report query with valid templating returns nil and the expected query output",
+			reportTemplate: &ReportQueryTemplateContext{
+				ReportQuery:       newTestReportQuery(ds1, "SELECT * FROM {| dataSourceTableName .Report.Inputs.UnreadyDeploymentReplicas |}"),
+				ReportDataSources: []*metering.ReportDataSource{ds1},
+				PrestoTables: []*metering.PrestoTable{
+					newTestPrestoTable(ds1.Status.TableRef.Name, testNamespace, testNamespace, nil),
+				},
+			},
+			templateContext: TemplateContext{
+				Report: ReportTemplateInfo{
+					Inputs: map[string]interface{}{"UnreadyDeploymentReplicas": ds1.Name},
+				},
+			},
+			expectOutput: fmt.Sprintf("SELECT * FROM hive.default.%s", ds1.Status.TableRef.Name),
+		},
+		{
+			name: "invalid report query with invalid templating (missing Inputs field) returns error and the expected error message",
```

Comment:
You can probably remove the ` and the expected error message` from all the names